c/polyjuice.h

```diff
@@ -567,20 +577,24 @@ struct evmc_result call(struct evmc_host_context* context,
                    msg->flags == EVMC_STATIC,
                    msg->input_data, msg->input_size,
                    (uint8_t**)&res.output_data, &res.output_size);
+    if (is_fatal_error(ret)) {
+      context->error_code = ret;
+    }
     if (ret != 0) {
       debug_print_int("call pre-compiled contract failed", ret);
-      context->error_code = ret;
-      res.status_code = EVMC_REVERT;
+      res.status_code = EVMC_INTERNAL_ERROR;
       return res;
     }
     res.status_code = EVMC_SUCCESS;
     debug_print_data("output data", res.output_data, res.output_size);
   } else {
     ret = handle_message(gw_ctx, context->from_id, context->to_id, &context->destination, msg, &res);
+    if (is_fatal_error(ret)) {
+      context->error_code = ret;
+    }
     if (ret != 0) {
       ckb_debug("inner call failed (transfer/contract call contract)");
-      context->error_code = ret;
-      res.status_code = EVMC_REVERT;
+      res.status_code = EVMC_INTERNAL_ERROR;
```

Comment:
What happend if we failed to call another contract? shouldn't it be a EVMC_REVERT status?